### ISSUE TITLE
JSON substitution is case sensitive by default, and also does not log missed keys that match apart from case

### ISSUE THREAD

**Issue by hoffmanntravis** [microsoft/azure-pipelines-tasks#13441]
This is somewhat similar to the issue below, but my point is more of a difference in design opinion than a bug:
https://github.com/microsoft/azure-pipelines-tasks/issues/6656

Relevant code to the best of my knowledge, or in the general vicinity:
https://github.com/microsoft/azure-pipelines-tasks/blob/a6f349070b6ed807a134974af44e49cf262b8037/Tasks/Common/webdeployment-common-v2/jsonvariablesubstitutionutility.ts#L71

Comment from 6656 that I have found to be true:
'JSON variable substitutions are case-sensitive. Ensure that the casing provided in the variable and appsettings.json are same.'

First, I disagree with this decision as a default.  I can't think of many, if any, cases where I would want to differentiate two json elements by case within the same level.  True, JSON _is_ case sensitive and it does allow for two keys to be differentiated at the same level based on case.  But is this really something a user would ever intend to do?  Perhaps, but probably not.  Which thus leads me to my next point.

If case sensitive vars are unlikely to be intentional, the default should be case insensitive with case sensitive as an override option.  However, this is a breaking change and so on.  If we don't agree here, a log statement would help quite a bit to mitigate this.  "The following pattern was matched, but JSON variable substitution is case sensitive.  Skipping this key."

One could arguably blame the user for making the case mistake, but I would offer the counterpoint that good UX / DX design should strive to match intent of the user and warn them when there may be a mismatch between intent and results.

Thanks,
Travis

**Comment by jhitze** [comment]
I agree that there should at least be some kind of log output for keys that were ignored or closely matched but missed. It would really help in debugging issues.
